src/machines/java/com/enderio/machines/client/gui/widget/ActiveWidget.java

```diff
@@ -0,0 +1,69 @@
+package com.enderio.machines.client.gui.widget;
+
+import com.enderio.EnderIO;
+import com.enderio.core.client.gui.widgets.EIOWidget;
+import com.enderio.machines.common.blockentity.MachineState;
+import com.enderio.machines.common.blockentity.MachineStateType;
+import com.enderio.machines.common.lang.MachineLang;
+import com.mojang.blaze3d.systems.RenderSystem;
+import net.minecraft.client.gui.GuiGraphics;
+import net.minecraft.client.gui.narration.NarrationElementOutput;
+import net.minecraft.client.gui.screens.Screen;
+import net.minecraft.network.chat.Component;
+import net.minecraft.resources.ResourceLocation;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Supplier;
+
+public class ActiveWidget extends EIOWidget {
+    protected static final ResourceLocation WIDGETS = EnderIO.loc("textures/gui/40/widgetsv2.png");
+
+    private final Screen displayOn;
+    private final Supplier<Set<MachineState>> state;
+
+    public ActiveWidget(Screen displayOn, Supplier<Set<MachineState>> state, int x, int y) {
+        super(x, y, 16, 16);
+        this.displayOn = displayOn;
+        this.state = state;
+    }
+
+    @Override
+    protected void renderWidget(GuiGraphics guiGraphics, int mouseX, int mouseY, float partialTick) {
+        RenderSystem.defaultBlendFunc();
+        RenderSystem.enableDepthTest();
+        MachineState prio = null;
+        for (MachineState machineState: state.get()) {
+            if (prio == null || machineState.type().getPriority() > prio.type().getPriority()) {
+                prio = machineState;
+            }
+        }
+        guiGraphics.blit(WIDGETS, x, y, 0, prio == null ? 16 : prio.type().getPriority()*16, 28*16, width, height, 256, 256);
+
+        //Stopped
```

Comment:
Are these leftover/needing removed or is this a future addition to the feature? Can we remove or comment as necessary please :)